python/tank/deploy/tank_commands/setup_project_params.py

```diff
@@ -1194,7 +1199,33 @@ def get_uri(self):
         :returns: string
         """
         return self._config_uri
-        
+
+    def is_project_config(self):
+        """
+        Returns if the configuration is from a configured project.
+
+        :returns: True if the configuration is from a project, False if it comes from the AppStore, GitHub or a zip
+                  file.
+        """
+        return self._config_mode == self._PROJECT_CONFIG
+
+    def get_pipeline_configuration(self):
+        """
+        Resolves the potential pipeline configuration based on the configuration uri. Potential is employed here because
+        there's no guarantee this folder is actually part of a pipeline configuration.
+
+        :returns: Path to the pipeline configuration associated with the configuration uri.
+
+        :raises TankError: This exception is raised when the configuration was pulled from GitHub, AppStore or zip file,
+            since no pipeline configuration can be associated with these.
+        """
+        if not self.is_project_config():
+            raise TankError("Pipeline configuration uri only exists for configured projects!")
```

Comment:
Polish: 

How about including the uri here just to help an error/support situation? E.g. something like

```
"Cannot resolve pipeline configuration for '%s', because it doesn't belong to an existing project!" % self._config_uri
```

What do you think?